libcontainer/factory_linux.go

```diff
@@ -26,7 +26,7 @@ const (
 )
 
 var (
-	idRegex  = regexp.MustCompile(`^[\w_-]+$`)
+	idRegex  = regexp.MustCompile(`^[\w-\.]+$`)
```

Comment:
FWIW, I'm fairly sure most regex implementations consider `[.]` to mean the literal `.` character (not "any" character, since `[.]` would be meaningless otherwise).